File: C++/Merge.cpp
#include<iostream>

struct ListNode {
    int val;
    struct ListNode *next;
    ListNode(int x) : val(x), next(nullptr) {}
};

class Solution{
    public:
        ListNode* Merge(ListNode* pHead1, ListNode* pHead2) {
            if(pHead1 == nullptr){
                return pHead2;
            }
            if(pHead2 == nullptr){
                return pHead1;
            }

            ListNode* p = new ListNode(0);
            ListNode* current =p;
            while(pHead1 !=nullptr && pHead2 != nullptr){
                if(pHead1->val <= pHead2->val){
                    current->next = pHead1;
                    pHead1 = pHead1->next;
                }
                else{
                    current->next = pHead2;
                    pHead2 = pHead2->next;
                }
                current =current->next;
            }
            if(pHead1 !=nullptr){
                current->next =pHead1;
            }
            if(pHead2 !=nullptr){
                current->next =pHead2;
            }
            return p->next;
        }
};

int main(){
    ListNode *pHead1=new ListNode(1);
    pHead1->next=new ListNode(3);
    pHead1->next->next=new ListNode(5);
    ListNode *pHead2 =new ListNode(2);
    pHead2->next=new ListNode(4);
    pHead2->next->next=new ListNode(6);
    Solution solution;

    std::cout<<"Original list1:";
    for(ListNode* node=pHead1;node!=nullptr;node=node->next){
        std::cout<<node->val<<" ";
    }
    std::cout<<std::endl;

    std::cout<<"Original list2:";
    for(ListNode* node=pHead2;node !=nullptr;node=node->next){
        std::cout<<node->val<<" ";
    }
    std::cout<<std::endl;

    ListNode* mergedHead = solution.Merge(pHead1, pHead2);
    std::cout<<"Merged list:";
    for(ListNode* node=mergedHead;node !=nullptr;node = node->next){
        std::cout<<node->val<<" ";
    }
    std::cout<<std::endl;

    while(pHead1){
        ListNode* next =pHead1->next;
        delete pHead1;
        pHead1=next;
    }
    while(pHead2){
        ListNode* next=pHead2->next;
        delete pHead2;
        pHead2=next;
    }

    return 0;
}

File: python/Merge.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None
        
class Solution:
    def Merge(self , pHead1: ListNode, pHead2: ListNode) -> ListNode:
        if pHead1 is None:
            return pHead2
        if pHead2 is None:
            return pHead1
        p=ListNode(0)
        current =p
        while pHead1 is not None and pHead2 is not None:
            if pHead1.val <= pHead2.val:
                current.next =pHead1
                pHead1=pHead1.next
            else:
                current.next = pHead2
                pHead2=pHead2.next
            current =current.next
        if pHead1 is not None:
            current.next =pHead1
        if pHead2 is not None:
            current.next =pHead2
        return p.next
    
def print_list(node):
    while node:
        print(node.val, end=" -> ")
        node = node.next
    print("None")

list1 = ListNode(1)
list1.next = ListNode(3)
list1.next.next = ListNode(5)

list2 = ListNode(2)
list2.next = ListNode(4)
list2.next.next = ListNode(6)

print("List 1:")
print_list(list1)
print("List 2:")
print_list(list2)

solution = Solution()
merged_head = solution.Merge(list1, list2)

print("Merged List:")
print_list(merged_head)